### PR TITLE
Update some dependencies

### PR DESCRIPTION
**Pull request by 8cAyqpVKio**
Update some dependencies:

Gradle, from 7.3.3 to 8.2.1
Android Gradle Plugin, from 7.2.2 to 8.0.2
Kotlin, from 1.8.10 to 1.8.22
maven-publish is now support by default
compose-compiler, from 1.4.3 to 1.4.8

Others:

Update .gitignore, and remove .idea/**
(build.gradle) Add namespace, required by agp 8.x
(build.gradle) Migrate from buildscript.dependencies to plugins
(build.gradle, settings.gradle) Refactor repositories
(jitpack.yml, ci) Update Java, from 11 to 17, required by agp 8.x

### CODE REVIEW

**Comment by 8cAyqpVKio** [comment]
jitpack.io CI is confirmed to work([my fork's jitpack.io build log](https://jitpack.io/com/github/8cAyqpVKio/LazyColumnScrollbar/f02ea99af0/build.log)).
I'll leave it up to you to update other dependencies(compose-bom, etc).

File: .github/workflows/default.yml [hunk 1]
```diff
@@ -9,24 +9,28 @@ on:
   push:
     branches:
       - master
+      - dev
+      - '*-dev'
+      - 'dev-*'
```
Comment (nanihadesuka) [inline .github/workflows/default.yml]: Remove these ones. Only want trigger for master and PR

File: .github/workflows/default.yml [hunk 2]
```diff
@@ -9,24 +9,28 @@ on:
...
 
 jobs:
   runLibUnitTests:
     name: Run lib unit tests
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```
Comment (nanihadesuka) [inline .github/workflows/default.yml]: I prefer a not moving target version. Can you add a fixed version?

File: .gitignore [hunk 3]
```diff
@@ -1,15 +1,144 @@
+# Created by https://www.toptal.com/developers/gitignore/api/linux,macos,windows,gradle,android
+# Edit at https://www.toptal.com/developers/gitignore?templates=linux,macos,windows,gradle,android
```
Comment (nanihadesuka) [inline .gitignore]: remove these comments

File: .gitignore [hunk 4]
```diff
@@ -1,15 +1,144 @@
...
+# Eclipse Gradle plugin generated files
+# Eclipse Core
+.project
+# JDT-specific (Eclipse Java Development Tools)
+.classpath
+
+### Gradle Patch ###
+# Java heap dump
+
+# End of https://www.toptal.com/developers/gitignore/api/linux,macos,windows,gradle,android
```
Comment (nanihadesuka) [inline .gitignore]: Same remove.

As for the rest of the lines. I guess this cover all the possible use cases / platforms. Ok by me.

File: app/build.gradle [hunk 5]
```diff
@@ -36,7 +37,7 @@ android {
         compose true
     }
     composeOptions {
-        kotlinCompilerExtensionVersion "1.4.3"
+        kotlinCompilerExtensionVersion compose_compiler_version
```
Comment (nanihadesuka) [inline app/build.gradle:40]: If you want to go an extra mile you can move all the Deps and versions to a buildSrc and convert the groovy setup to gradle kotlin

File: jitpack.yml [hunk 6]
```diff
@@ -1,2 +1,5 @@
-jdk:
-  - openjdk11
\ No newline at end of file
+before_install:
+  - source "$HOME/.sdkman/bin/sdkman-init.sh"
+  - sdk update
+  - sdk install java 17.0.7-tem
+  - sdk use java 17.0.7-tem
```
Comment (nanihadesuka) [inline jitpack.yml:5]: This looks like something custom for your machine (sdk is not installed by default). Revert and use `- openjdk17`

File: settings.gradle [hunk 7]
```diff
@@ -1,3 +1,20 @@
+pluginManagement {
+    repositories {
+        google()
+        mavenCentral()
+        gradlePluginPortal()
+    }
+}
+
+dependencyResolutionManagement {
+    repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
```
Comment (nanihadesuka) [inline settings.gradle]: Reading the docs this line means the gradle sync will fail if there are repositories outside of setting.gradle. Looks ok to me.

**Review by nanihadesuka — CHANGES_REQUESTED** [review]
Revert the jitpack config setup changes (sdk call). The rest is optional but prefered

**Comment by nanihadesuka** [comment]
Btw. I didn't update the compose bom because it has many breaking changes. Will probably do a major release for the bom change.

**Comment by 8cAyqpVKio** [comment]
Thank you for your replies.
I’ll check each one.

File: .gitignore (hunk 4 above)
Comment (8cAyqpVKio) [reply .gitignore]: First of all, I forgot to mention something important, important.
This gitignore is generated by gitignore.io and is under CC0.
https://github.com/toptal/gitignore.io#license-of-the-generated-files

So it's licensed safe.

BTW, slimming down gitignore based on your suggestion.

File: jitpack.yml (hunk 6 above)
Comment (8cAyqpVKio) [reply jitpack.yml:5]: In jitpack.io CI server, jdks managed by sdkman.
https://jitpack.io/docs/BUILDING/

If you want to set JDK17, we must set jdk version by using sdkman commands.
https://github.com/jitpack/jitpack.io/issues/4778

So this, jitpack.yml should be fine.

File: app/build.gradle (hunk 5 above)
Comment (8cAyqpVKio) [reply app/build.gradle:40]: I will stop this time. It's a big, unimportant refactoring.

**Comment by 8cAyqpVKio** [comment]
jitpack.yml is ok. https://github.com/nanihadesuka/LazyColumnScrollbar/pull/19#discussion_r1264439704

I checked all, so please check again

**Pull request merged**